### DISCUSSION TITLE
How to add a custom static method?

### DISCUSSION THREAD

**Discussion by blueridgemountains1** [spatie/laravel-data#576] [General]
I migrated from the dto package to the laravel-data package.

I want a custom static method 
```php
<?php

class PostmarkResponse extends Data
{

    public function __construct(
        public string $email,
        public string $status,
        public ?string $message = null,
        public ?string $stream = null,
    ) {
    }

    public static function fromSingleResponse(array $response): static
    {
        return static::from([
            'email' => $response['EmailAddress'],
            'status' => $response['Status'],
            'message' => $response['Message'] ?? null,
            'stream' => $response['Stream'] ?? null,
        ]);
    }

}

```

However, I cannot call `PostmarkResponse::fromSingleResponse` because I get the error `TypeError: Spatie\LaravelData\Resolvers\DataFromSomethingResolver::createFromCustomCreationMethod(): Return value must be of type ?Spatie\LaravelData\Contracts\BaseData, Illuminate\Support\Collection returned`

How can I create a custom static method without the `createFromCustomCreationMethod` method firing?

**Comment by rubenvanassche** [comment]
Not entirely sure how a collection is returned?

As for not automatically calling methods, we only do this with static methods starting with from. So, creating a method called createFromSomething() should work.
